esp-wifi/src/wifi/os_adapter.rs

```diff
@@ -24,7 +28,12 @@ use crate::{
 #[cfg(feature = "esp32c3")]
 use crate::compat::common::syslog;
 
-pub static mut WIFI_STATE: i32 = -1;
+use super::WifiEvent;
+
+pub(crate) static mut WIFI_STATE: i32 = -1;
+
+pub(crate) static WIFI_EVENTS: Mutex<RefCell<EnumSet<WifiEvent>>> =
```

Comment:
Should we make this cfg'd away in non async? Part of me tells me it might actually be quite helpful, but we probably shouldn't expose it unless we actually have a use for it else where.

Reply:
Maybe we can keep it unconditionally and have a comment